File: 0401_0500/0440/0440.py
def dist(a, b):
    return ((a[0] - b[0]) * (a[0] - b[0]) + (a[1] - b[1]) * (a[1] - b[1])) ** 0.5

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    m = [[25 * i, 0, False] for i in range(5)]
    d = 0
    for s in f.readlines():
        a = [int(x) for x in s.split()]
        for x in m:
            if not x[2] and dist(a, x) <= 10:
                x[2] = True
                d += 1
                break
    q.write(str(d))
                

File: 0401_0500/0440/0440.c
#include <stdio.h>
#include <math.h>

typedef struct {int x, y;} dot;

double distance(dot a, dot b)
{
	return sqrt((double)((a.x - b.x) * (a.x - b.x) + (a.y - b.y) * (a.y - b.y)));
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int t[5] = {0}, s = 0;
	dot m[5] = { {0, 0}, {25, 0}, {50, 0}, {75, 0}, {100, 0} }, a;

	for (int i = 0; i < 5; ++i)
	{
		fscanf(f, "%d%d", &a.x, &a.y);
		for (int j = 0; j < 5; ++j)
			if (!t[j] && (t[j] = (distance(m[j], a) <= 10)) && ++s)
				break;
	}
	fprintf(q, "%d", s);
	return 0;
}
